fastapi/routing.py

```diff
@@ -490,7 +490,9 @@ def __init__(
         self.name = get_name(endpoint) if name is None else name
         self.path_regex, self.path_format, self.param_convertors = compile_path(path)
         if methods is None:
-            methods = ["GET"]
+            methods = ["GET", "HEAD"]
+        elif "GET" in methods:
+            methods = set(methods).union({"HEAD"})
```

Comment:
Mixing lists and sets doesn't look great ... I would prefer a consistent processing approach and stick to sets.